### ISSUE TITLE
Support for App Service IP Restrictions Header Filter Settings

### ISSUE THREAD

**Issue by DThrasher** [terraform-providers/terraform-provider-azurerm#11454]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Microsoft Azure recently implemented support for restricting access to a particular Front Door instance through the use of a new HTTP header:  X-Azure-FDID. See these posts for details:
https://docs.microsoft.com/en-us/azure/frontdoor/front-door-http-headers-protocol
https://docs.microsoft.com/en-us/azure/app-service/app-service-ip-restrictions#restrict-access-to-a-specific-azure-front-door-instance
https://henrihietala.fi/limit-access-to-your-azure-web-app-from-your-azure-front-door-only/

The Terraform AzureRM App Service resource allows you to set ip_restrictions, but not header filter settings values  

### New or Affected Resource(s)

* azurerm_app_service

### Potential Terraform Configuration

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key.
```

### References

* #0000

**Comment by aristosvo** [comment]
Hi @DThrasher !

I believe this is a duplicate of #11156, and probably solved through unmerged PR #11209

**Comment by katbyte** [comment]
Thanks @aristosvo - looks like your correct so i'm going to close this. If we're incorrect @DThrasher please do let us know!
